docs/design/fabric-v2+/trusted_ledger.md

```diff
@@ -0,0 +1,162 @@
+- keep redundant validation code (in stock peer and TLCC) in sync
+- keep code relation traceable and changes trackable
+- consistent naming and code structure as much as possible
+- bonus: 
+    - automated code-changes notification; notifies whenever relevant Fabric code changes and might FPC must be updates.
+
+### Non-features
+
+- No state-based endorsement
+- No custom MSP (no idemix)
+- No support for custom endorsing/validation plugins for non-FPC and FPC chaincodes
+- Authentication and decorators
+- TODO check for more non-features in RFC
+
+## Design
+
+### Chaincode execution support
+    - secure channel
+    - validate proposal
+        - check org is "write"
+        - replay protection
+
+### Ledger maintainance
+
+Channel configuration:
+- Parse channel definition
+    - parse consensus definition
+- parse channel reader/writers
+- validate block signatures (Note that this check implies correctness as onfig blocks are validated by the ordering service ([see code](https://github.com/hyperledger/fabric/blob/f27912f2f419c3b35d2c1df120f19585815eceb0/orderer/common/msgprocessor/standardchannel.go#L131)))
+- Maintain msp metadata for signature validation
+- parse Lifecycle policy 
+    - validate Lifecycle policy
+    We only support: a majority of Org.member (See [Link](https://hyperledger-fabric.readthedocs.io/en/release-2.2/chaincode_lifecycle.html#install-and-define-a-chaincode))
+
+Access control:
+- delegate this "reader/writer/admin" check to admins;
+- TLCC only verifies endorsement policies and thereby implicitly the "writers" check is performed by the endorsers
+
+MSP:
+- One Org - One MSP mapping - One root cert
+- Implement X509-based MSP
+    - Restrict:
+        - no intermediate
+        - no CRLs
+        - only support member role
+    - any certificate will match to role member
+
+Endorsing:
+- Phase1: Designated enclave only
+- Phase2: Any enclave that runs a particular FPC chaincode
+
+Versioning:
+- single autonomously monotonously increasing version number??
+
+Non-FPC Validation:
+- Transaction submitter identity validation
+    - submitter satisfies channel's writes policy
+- ERCC endorsement signatures verification
+- ERCC endorsement policy validation
+    - we only support (explicityly) Majority endorsement policy
+    - Restrict to ERCC namespace only
+
+FPC Validation:
+- Introduce FPC transaction type (similar to introduce FPC namespace) and create
+a dedicated FPC tx processor; (removes the need of custom validation plugins and interference with existing Fabric validation logic; and also gives more freedom to FPC validation logic as no it not longer bound to the structure and format of endorsement transaction).
+
+- Support for (subset of) endorsing policies
+    - FPC Chaincode (see above)
+
+- Parse chaincode definitions
+- Transaction submitter identity validation
+    - submitter satisfies channel's writes policy
+
+- FPC endorsement policy validation
+    - Support only: ANY
```

Comment:
Not sure what you mean by ANY. in https://github.com/hyperledger-labs/fabric-private-chaincode/blob/master/docs/design/fabric-v2%2B/fpc-management.md we say an OR with a single org (and from above, we would even further restrict it that the role must be `member`; in that document we would also allow other roles ...)